🚀[Release v3.13.5] Merge into Develop

kcw-grunt opened the pull request:
# Overview
Added a new logic to block non supported countries from ever seeing Moonpay
- 3 weeks later
- Code improvements
- New localized push notifications: Push notifications in your set language in Litewallet

## Bugfixes:
- Fix crashed due to sqlite writes
- Simplified and localized sync views

mosadialiou commented on loafwallet/src/Strings/da.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d af %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/de.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d von %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/es.lproj/Localizable.strings:
```diff
@@ -1357,3 +1357,9 @@
 "WritePaperPhrase.step" = "%1$d de %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/fr.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d de %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/id.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d dari %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/it.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d di %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/ja.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%2$dの%1$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/ko.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%2$d의 %1$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/nl.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d van %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/pt.lproj/Localizable.strings:
```diff
@@ -1357,3 +1357,9 @@
 "WritePaperPhrase.step" = "%1$d de %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/ru.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d из %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/sv.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d av %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/zh-Hans.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%1$d / %2$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/src/Strings/zh-Hant.lproj/Localizable.strings:
```diff
@@ -1356,3 +1356,9 @@
 "WritePaperPhrase.step" = "%2$d 之 %1$d";
+
+/* Moonpay buy financial details */
+"BuyCenter.moonpayFinancialDetails" = "";
```
missing translations

mosadialiou commented on loafwallet/BuyWKWebViewController.swift:
```diff
@@ -55,33 +49,50 @@ class BuyWKWebViewController: UIViewController, WKNavigationDelegate, WKScriptMe
         let request = URLRequest(url: url)
-        wkWebView.load(request)
+
+        switch partnerName {
```
Why using `partnerName` since this class is only for `Simplex` partner?
I think you should delete all references to another partner and just assume it's just Simplex until it's used for another one.

kcw-grunt replied:
good point

mosadialiou commented on loafwallet/BuyWKWebViewController.swift:
```diff
@@ -17,34 +19,26 @@ class BuyWKWebViewController: UIViewController, WKNavigationDelegate, WKScriptMe
+    var didDismissChildView: (() -> ())?
     
-    private let wkProcessPool = WKProcessPool()
+    var partnerName: PartnerName?
```
`partnerName` is not used! so it can be removed.

kcw-grunt replied:
🧹

mosadialiou commented on loafwallet/BuyTableViewController.swift:
```diff
@@ -7,22 +7,53 @@
 
-class BuyTableViewController: UITableViewController { 
-     
+enum PartnerName {
```
It seems like this enum is only used by Simplex. We can remove it.

kcw-grunt replied:
🎉🤔🧹

mosadialiou reviewed (approved):
LGTM

kcw-grunt commented:
## Test Staging Screenshots
|Success|Failed|
|![moonpaysuccess-stage](https://user-images.githubusercontent.com/2899463/113561117-ff732c00-95fb-11eb-91a2-2461a26be94e.gif)|![moonpay-fail-stage](https://user-images.githubusercontent.com/2899463/113561228-29c4e980-95fc-11eb-9ed0-bb5bc997f29a.gif)|

kcw-grunt committed:
version bump

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

kcw-grunt committed:
removed unused code

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

kcw-grunt committed:
Added Ukrainian localizations

- version bump

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

kcw-grunt committed:
Disabled bitrefill until compliance docs received

Adjusted low fees tier

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

kcw-grunt committed:
version bump

- Removed Unstoppable code
- Uncommented out Bitrefill
- added more branding

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

kcw-grunt committed:
bugfix of SyncProgress crash

https://console.firebase.google.com/project/litewallet-beta/crashlytics/app/ios:com.litecoin.loafwallet/issues/fecabb7c7120dca59920bd7a1918c710?time=last-seven-days&types=crash&sessionEventKey=040a831703ba417ab9c755d44759ced2_1949520751979676838
- Removed Unstoppable code
- Uncommented out Bitrefill
- added more branding
- Need to redesign Moonpay for Apple meeting
- Added labels

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

Further polishing the new buy view

- rebuilt moonpay view
- removed unused logos
- added paste button
- improved layout
- Added non-buy country tabBar ViewController
- Fixed the region filter to change tabbar for unsupported countries
- Added unsupported event for Firebase
- Reset sync params to original value

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

Refactored code

Signed-off-by: kcw-grunt <mrkerrywashington@icloud.com>

josikie reviewed (approved):
Cool! Thank you! @kcw-grunt
